Mælaborð.sql

```diff
@@ -0,0 +1,574 @@
+        Garún AS Verð
+    FROM Bjór WHERE Garún IS NOT NULL
+
+    UNION ALL
+
+    SELECT 
+        Bar,
+        "Stærð (mL)",
+        Latitude,
+        Longitude,
+        "Mynd (url)",
+        'Helga' AS Bjór,
+        Helga AS Verð
+    FROM Bjór WHERE Helga IS NOT NULL
+
+    UNION ALL
+
+    SELECT 
+        Bar,
+        "Stærð (mL)",
+        Latitude,
+        Longitude,
+        "Mynd (url)",
+        'Snorri' AS Bjór,
+        Snorri AS Verð
+    FROM Bjór WHERE Snorri IS NOT NULL
+
+    UNION ALL
+
+    SELECT 
+        Bar,
+        "Stærð (mL)",
+        Latitude,
+        Longitude,
+        "Mynd (url)",
+        'Úlfrún' AS Bjór,
+        Úlfrún AS Verð
+    FROM Bjór WHERE Úlfrún IS NOT NULL
+
+    UNION ALL
+
+    SELECT 
+        Bar,
+        "Stærð (mL)",
+        Latitude,
+        Longitude,
+        "Mynd (url)",
+        'Úlfur' AS Bjór,
+        Úlfur AS Verð
+    FROM Bjór WHERE Úlfur IS NOT NULL
+
+    UNION ALL
+
+    SELECT 
+        Bar,
+        "Stærð (mL)",
+        Latitude,
+        Longitude,
+        "Mynd (url)",
+        'Guinness' AS Bjór,
+        Guinness AS Verð
+    FROM Bjór WHERE Guinness IS NOT NULL
+
+    UNION ALL
+
+    SELECT 
+        Bar,
+        "Stærð (mL)",
+        Latitude,
+        Longitude,
+        "Mynd (url)",
+        'Bóndi' AS Bjór,
+        Bóndi AS Verð
+    FROM Bjór WHERE Bóndi IS NOT NULL
+
+    UNION ALL
+
+    SELECT 
+        Bar,
+        "Stærð (mL)",
+        Latitude,
+        Longitude,
+        "Mynd (url)",
+        'Heimabrugg/Annar IPA' AS Bjór,
+        "Heimabrugg/Annar IPA" AS Verð
+    FROM Bjór WHERE "Heimabrugg/Annar IPA" IS NOT NULL
+) ORDER BY Bar, Bjór;
+
+-- Aukalínum eytt
+DELETE FROM Bjórkort WHERE Verð IS NULL OR trim(Verð) = '';
+DELETE FROM Bjórkort WHERE Bar IS 'Bar' OR trim(Bar) = 'Bar';
+
+------------------------------------------------------------------------------------------------
+-- 6
+
+-- Dálkum með meðalverðum bætt í Bjór Vínbúð töflu
+------------------------------------------------------------------------------------------------
+
+-- Dálk bætt við með meðalverðum á bjór í bænum
+ALTER TABLE Bjór_Vínbúðin ADD COLUMN "Meðalverð í bæ (Kr)" INTEGER;
+UPDATE Bjór_Vínbúðin
+SET "Meðalverð í bæ (Kr)" = (
+    SELECT CAST(ROUND(AVG(Verð)) AS INTEGER)
+    FROM Bjórkort
+    WHERE Bjórkort.Bjór = Bjór_Vínbúðin.Bjór
+);
+
+-- Dálk bætt við með meðal lítraverðum á bjórum í bænum
+ALTER TABLE Bjór_Vínbúðin ADD COLUMN "Meðal lítraverð í bæ (Kr)" INTEGER;
+UPDATE Bjór_Vínbúðin
+SET "Meðal lítraverð í bæ (Kr)" = (
+    SELECT CAST(ROUND("Meðalverð í bæ (Kr)" / (AVG("Stærð (mL)") / 1000)) AS INTEGER)
+    FROM Bjórkort
+    WHERE Bjórkort.Bjór = Bjór_Vínbúðin.Bjór
+);
+
+-- Dálk bætt við með meðal lítraverðum á bjórum í vínbúðinni
+ALTER TABLE Bjór_Vínbúðin ADD COLUMN "Lítraverð (Kr)" INTEGER;
+UPDATE Bjór_Vínbúðin
+SET "Lítraverð (Kr)" = ("Verð (Kr)" / ("Stærð (mL)" / 1000.0))
+WHERE "Stærð (mL)" IS NOT NULL AND "Stærð (mL)" > 0;
```

Comment:
Það væri í raun betra að gera frekar sýndartöflu fyrir þetta.
```suggestion
CREATE VIEW v_Bjór_Vínbúðin AS
SELECT 
    Bjór_Vínbúðin.*,
    Útreikningar."Meðalverð í bæ (Kr)",
    Útreikningar."Meðal lítraverð í bæ (Kr)",
    ("Verð (Kr)" / ("Stærð (mL)" / 1000.0)) AS "Lítraverð (Kr)"
FROM Bjór_Vínbúðin
LEFT JOIN Útreikningar ON Bjór_Vínbúðin.Bjór = Útreikningar.Bjór;
```
eða þá gera eitt UPDATE sem tekur alla dálka fyrir í einu:
```suggestion
WITH Útreikningar AS (
    SELECT
        Bjór_Vínbúðin.Bjór,
        CAST(ROUND(AVG(Bjórkort.Verð)) AS INTEGER) AS "Meðalverð í bæ (Kr)",
        CAST(ROUND(AVG(Bjórkort.Verð) / (AVG(Bjórkort."Stærð (mL)") / 1000)) AS INTEGER) AS "Meðal lítraverð í bæ (Kr)"
    FROM Bjórkort
    JOIN Bjór_Vínbúðin ON Bjórkort.Bjór = Bjór_Vínbúðin.Bjór
    GROUP BY Bjór_Vínbúðin.Bjór
)
UPDATE Bjór_Vínbúðin
SET 
    "Meðalverð í bæ (Kr)" = Útreikningar."Meðalverð í bæ (Kr)",
    "Meðal lítraverð í bæ (Kr)" = Útreikningar."Meðal lítraverð í bæ (Kr)",
    "Lítraverð (Kr)" = ("Verð (Kr)" / ("Stærð (mL)" / 1000.0))
FROM Útreikningar
WHERE Bjór_Vínbúðin.Bjór = Útreikningar.Bjór
  AND "Stærð (mL)" IS NOT NULL AND "Stærð (mL)" > 0;
```